src/main/scala/com/cloudant/clouseau/ExternalSnapshotDeletionPolicy.scala

```diff
@@ -0,0 +1,70 @@
+package com.cloudant.clouseau
+
+import java.io.File
+import java.nio.file.StandardCopyOption.ATOMIC_MOVE
+import java.nio.file.Files
+import java.util.List
+import java.util.UUID
+import org.apache.lucene.index.IndexCommit
+import org.apache.lucene.index.IndexDeletionPolicy
+import org.apache.lucene.store.FSDirectory
+import scala.collection.JavaConversions._
+
+class ExternalSnapshotDeletionPolicy(dir: FSDirectory) extends IndexDeletionPolicy {
```

Comment:
I'm reading up on the old blogs on how to create snapshots. I know IndexDeletionPolicy is in memory only, which doesn't help us as we want a copy of the actual disk files. Can you explain the rationale behind creating a new extended class vs using https://lucene.apache.org/core/4_6_1/core/org/apache/lucene/index/PersistentSnapshotDeletionPolicy.html?
I do notice you're creating temporary directory names so perhaps that's the reason.

Reply:
Because PersistentSnapshotDeletionPolicy creates the snapshot inside the Lucene directory (i.e, under /srv/search_index/... for cloudant), and the list of files constituting the snapshot is written in a binary file (snapshots_X) which OS-level tools can't read.

"IndexDeletionPolicy is in memory only" is not right. The commits that are preserved in an index are durable, it's just the default policy (KeepOnlyLastCommitDeletionPolicy) that chooses to only preserve the latest one. A policy could easily be written to preserve them all.

The purpose of this new Policy is to make it easy for administrators to take a consistent backup of the index. The directory that contains the snapshot is immutable once created (hence the atomic rename to your chosen name as the last step). The contents of that directory represent the index at the time the snapshot was created, they can be copied to some external source, and, when backup is complete, the snapshot directory can be deleted, which will potentially release the files in question (they might still be part of the _current_ state of the index).

Reply:
> "IndexDeletionPolicy is in memory only" is not right. The commits that are preserved in an index are durable, it's just the default policy (KeepOnlyLastCommitDeletionPolicy) that chooses to only preserve the latest one. A policy could easily be written to preserve them all.

My error, I meant to say SnapShotDeletionPolicy "only in memory":
https://lucene.apache.org/core/4_6_1/core/org/apache/lucene/index/SnapshotDeletionPolicy.html
"This class maintains all snapshots in-memory, and so the information is not persisted and not protected against system failures."
IIUC, if we use SnapShotDeletionPolicy then as long as the clouseau is up, lucene won't delete the commit files referenced by the snapshot. However, if clouseau crashed, then the snapshot will be lost and subsequently, the referenced commit files will eventually become stale and deleted. 
Anyways, it was for my own understanding only. Thanks for clarifying.

Reply:
>The contents of that directory represent the index at the time the snapshot was created, they can be copied to some external source, and, when backup is complete, the snapshot directory can be deleted, which will potentially release the files in question (they might still be part of the current state of the index).

Should we add a function to remove the snapshot/delete the files? Right now,  an operator would have to manually go in and delete the files (or write their own script to clean it up). It seems having a function would allow easier automation in the future so there aren't lingering snapshots around. I'm envisioning a scenario where snapshots are copied into cold storage but we forget to delete the snapshots on the nodes.

Reply:
aha, ok. yes, correct, SnapshotDeletionPolicy just maintains the list of IndexCommit's to preserve in memory, and all snapshots would be released on JVM exit. This is often fine, of course, and the benefit there is allowing a client to run queries over the same version of the index for a period of time, even if the index has subsequently changed.

Reply:
For cleanup, it is as simple as deleting the files in the directory specified in the {create_snapshot, Directory} location so can be left to the administrator.